test/extended/authorization/authorization.go

```diff
@@ -1241,3 +1242,93 @@ var _ = g.Describe("[sig-auth][Feature:OpenShiftAuthorization] authorization", f
+                err := oc.AsAdmin().WithoutNamespace().Run("patch").Args(
+                    "configs.imageregistry.operator.openshift.io/cluster",
+                    "--type=merge",
+                    "-patch", `{"spec": {"defaultRoute": true}}`,
+                ).Execute()
+                o.Expect(err).NotTo(o.HaveOccurred())
+
+                g.By("Wait for the image registry to become available")
+                available := isImageRegistryAvailable(oc)
+                o.Expect(available).To(o.BeTrue())
+
+                g.By("Grant public access to the openshift namespace")
+                defer oc.AsAdmin().WithoutNamespace().Run("policy").Args(
+                    "remove-role-from-group",
+                    "system:image-puller", "system:unauthenticated",
+                    "--namespace", "openshift",
+                ).Execute()
+                output, err := oc.AsAdmin().WithoutNamespace().Run("policy").Args(
+                    "add-role-to-group", "system:image-puller", "system:unauthenticated",
+                    "--namespace", "openshift",
+                ).Output()
+                o.Expect(err).NotTo(o.HaveOccurred())
+                o.Expect(output).To(o.ContainSubstring(
+                    "clusterrole.rbac.authorization.k8s.io/system:image-puller added: \"system:unauthenticated\"",
+                ))
+
+                g.By("Fetch image metadata")
+                output, err = oc.AsAdmin().Run("image").Args(
+                    "info", "--insecure",
+                    "registry.openshift-image-registry.svc.cluster.local/openshift/tools:latest",
+                    "--show-multiarch",
+                ).Output()
+                o.Expect(err).NotTo(o.HaveOccurred())
+                o.Expect(output).NotTo(o.ContainSubstring("error: unauthorized: authentication required"))
+                o.Expect(output).NotTo(o.ContainSubstring("Unable to connect to the server: no basic auth credentials"))
+                o.Expect(output).To(o.ContainSubstring(
+                    "registry.openshift-image-registry.svc.cluster.local/openshift/tools:latest",
+                ))
+            })
+        })
+    })
+})
+
+func isImageRegistryAvailable(oc *exutil.CLI) bool {
+    output, err := oc.AsAdmin().Run("get").Args(
+        "co", "image-registry",
+        "-o=jsonpath={.status.conditions[?(@.type=='Available')].status}",
+    ).Output()
+    o.Expect(err).NotTo(o.HaveOccurred())
+    return strings.TrimSpace(output) == "True"
+}
+
+func exposeRouteFromSVC(oc *exutil.CLI, rType, ns, route, service string) string {
+	err := oc.AsAdmin().WithoutNamespace().Run("create").Args("route", rType, route, "--service="+service, "-n", ns).Execute()
+	o.Expect(err).NotTo(o.HaveOccurred())
+	regRoute, err := oc.AsAdmin().WithoutNamespace().Run("get").Args("route", route, "-n", ns, "-o=jsonpath={.spec.host}").Output()
+	o.Expect(err).NotTo(o.HaveOccurred())
+	return regRoute
+}
+
+func waitRouteReady(route string) {
```

Comment:
Consider bringing [this code](https://github.com/openshift/cluster-image-registry-operator/blob/master/test/framework/route.go?plain=1#L17-L44) from the registry operator's own tests into this function - you'd need to adapt it slightly but it does what you want by watching the route object itself, which is perhaps slightly more elegant 😅